kernel/mlx_ethernet/src/command_queue.rs

```diff
@@ -771,3 +1744,209 @@ impl fmt::Debug for CommandInterfaceMailbox {
             .finish()
     }
 }
+
+/// Layout of output of command [`CommandOpcode::QueryHcaCap`]
+#[derive(FromBytes)]
+#[repr(C)]
+struct HCACapabilitiesLayout {
+    vhca_resource_manager:          Volatile<U32<BigEndian>>,
```

Comment:
are all of these volatiles necessary? seems odd to have so many volatile fields, and then a separate mirror struct without them. I'm puzzled here.

Reply:
Basically, `HCACapabilitiesLayout` gives the order of how different fields are arranged in memory when we are returned device capability information from the command `QueryHcaCap`. 
'HCACapabillites' is the actual values of these capabilities, which have to be extracted from subfields of each member of the `HCACapabilitiesLayout` struct. (see the get_capabilities() function).

I think the confusion is perhaps why I set each member always to a U32. It's because of the big endianness of the fields. If I use types smaller than u32, like u16 or u8, then I have to rearrange their order in the struct manually, and I believe that will lead to more confusion and chances of error.

But your pointing this out reminded me that all these fields only need to be ReadOnly. So thanks for that.